### ISSUE TITLE
Volume Formatting breaks Block mode in v2

### ISSUE THREAD

**Issue by m4eba** [hetznercloud/csi-driver#349]
Every volume gets formatted even if in block mode.

**Comment by timbrd** [comment]
This might be the cause why I am not able to provision a rook ceph cluster anymore.
When initializing the osds, rook does not find any valid block devices:

````
provision 2022-12-23 17:34:24.432561 I cephosd: skipping device "/mnt/set1-data-1tl48r" because it contains a filesystem "ext4"
provision 2022-12-23 17:34:24.461665 I cephosd: configuring osd devices: {"Entries":{}}                                                                                        
provision 2022-12-23 17:34:24.461685 I cephosd: no new devices to configure. returning devices already configured with ceph-volume.
````

The pvc configured is in block mode:

```
apiVersion: v1
kind: PersistentVolumeClaim
metadata:
  [...]
  name: set1-data-1tl48r
  namespace: rook-ceph
  [...]
spec:
  accessModes:
  - ReadWriteOnce
  resources:
    requests:
      storage: 90Gi
  storageClassName: hcloud-volumes
  volumeMode: Block
  volumeName: pvc-00bbd41c-548a-4d6d-a33d-05e0beba2584
status:
  accessModes:
  - ReadWriteOnce
  capacity:
    storage: 90Gi
  phase: Bound
```

**Comment by timbrd** [comment]
When I switch back to csi driver 1.6.0, I was able to provision a ceph cluster.

**Comment by m4eba** [comment]
I also run into this playing around with rook. What works for me is a simple if statement for the code that formats the volume: https://github.com/m4eba/csi-driver/commit/c3c8173013f306a8765c56fae5a9638c0b2dd17b

**Comment by Privatecoder** [comment]
this also breaks re-attaching LUKS encrypted volumes

**Comment by apricote** [comment]
Hey all, thanks for the report. I am currently working on this, the fix is ready but I want some proper tests around this to make sure that it will not break again in the future.

**Comment by apricote** [comment]
Fixed by #353, will be released in the next patch release.

**Comment by Privatecoder** [comment]
@apricote v2 (any version) does throw the error. Downgrading to 1.6.0 fixes it, however I‘d like to use a more recent version and not lose the pvcs.

**Comment by apricote** [comment]
It sounds like you setup the volumes with v1.6, but that version does not include support for encrypted volumes. Encrypted volumes were first released in v2.0.0. If your volumes are actually LUKS encrypted, attaching them with v1.6 should fail.

Can you describe exactly how and when you created these volumes, along with the version of the CSI driver that was installed at that point in time?
